### PR TITLE
GSoC-2022 latest_capability_matrix based on ValidatesRunner tests beam-2888

### PR DESCRIPTION
**Pull request by alive2020**
**Please** add a meaningful description for your change here
#2888
- According to ticket [Beam-2888](https://issues.apache.org/jira/browse/BEAM-2888): Created a completely new Capability Matrix that is based on the ValidatesRunner tests:
- Currently, the implementation includes: getting parsed JSON data from gcloud and rendering a small and big Capability Matrix table

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by alive2020** [comment]
@pabloem

**Comment by Abacn** [comment]
Thanks for the contribution. Could you please set `.gitignore` in your local "nodes_modules"? Those are JavaScript libraries and should not be contained in the Beam repository. Could consider providing a script to setup environment (not familiar with npm here but other people could help).

**Comment by pabloem** [comment]
ah I can review this btw. @alive2020 please do address @Abacn comment to remove nodes_modlues

File: .gitignore [hunk 1]
```diff
@@ -115,6 +115,7 @@ website/www/yarn-error.log
 
 # Node
 sdks/typescript/node_modules
+node_modules
```
Comment (Abacn) [inline .gitignore:118]:
```suggestion
node_modules/**
```

Comment (Abacn) [reply .gitignore:118]: Add this then run `git rm --cached -r node_modules` and commit the change. This will remove node_modules files from the branch.

**Comment by Abacn** [comment]
@alive2020 Please DO NOT put service account key into this public repo. They have been disabled once leaked.

File: .gitignore (hunk 1 above)
Comment (alive2020) [reply .gitignore]: Thank you. Removed.

**Comment by Abacn** [comment]
Will do

File: website/www/site/data/capability_matrix.yaml [hunk 2]
```diff
@@ -36,1463 +36,1463 @@ capability-matrix:
   categories:
     - description: What is being computed?
       anchor: what
-      color-y: "fff"
-      color-yb: "f6f6f6"
-      color-p: "f9f9f9"
-      color-pb: "d8d8d8"
-      color-n: "e1e0e0"
-      color-nb: "bcbcbc"
+      color-y: 'fff'
```
Comment (Abacn) [inline website/www/site/data/capability_matrix.yaml:39]: Is there a reason changing " -> '? They look redundant

File: .gitignore [hunk 3]
```diff
@@ -115,8 +115,11 @@ website/www/site/code_samples
 website/www/site/_config_branch_repo.toml
 website/www/yarn-error.log
 !website/www/site/content
+.test-infra/validate-runner/src/main/js/apache-beam-testing-keys.json
 
 # Node
+sdks/typescript/node_modules
+node_modules/**
```
Comment (Abacn) [inline .gitignore:122]: There are still multiple website dependency files in the repo root directory e.g. `package-locks.json`, `package.json`, `yarn.lock`. Should organize it in somewhere under `website` folder.

There is already a project root in "website/www/" , there is an `yarn.lock` and `package.json` file there for the website dependency. (and `website/www/node_modules` is already in .gitignore)

Or If it is necessary to create a different project to generete the capability matrix, could move them into `website/generate_capability_matrix` project or so. 

May ask @pabloem which one of the choice is preferred.

**Review by Abacn — CHANGES_REQUESTED** [review]
Thanks. Left some more comments about the organization.

File: website/www/site/data/capability_matrix.yaml (hunk 2 above)
Comment (alive2020) [reply website/www/site/data/capability_matrix.yaml:39]: I think, it's due to my vscode setup.

**Pull request closed without merging**